### PR TITLE
Initial implementation

### PR DESCRIPTION
**Pull request by simu**
Initial implementation which has the following:
* Jsonnet which configures deployment, services and ingress/route for acme-dns
* Jsonnet which configures backups  (with K8up and component-backup-k8up) for the sqlite DB used by acme-dns
* Documentation of component parameters
* How-to for setting up an acme-dns instance

## Checklist

- [x] Keep pull requests small so they can be easily reviewed. (AN: I tried :see_no_evil: )
- [x] Update the documentation.
- [x] Categorize the PR by setting a good title and adding one of the labels:
      `bug`, `enhancement`, `documentation`, `change`, `breaking`, `dependency`
      as they show up in the changelog
- [x] Rebase PR to clean up commits

### CODE REVIEW

File: component/main.jsonnet [hunk 1]
```diff
@@ -1,10 +1,319 @@
...
+        ],
+      },
+    }
+) + {
+  metadata+: {
+    annotations+: std.prune(params.api.ingress.annotations),
+  },
+};
+
+local makeNamespaced(key, obj) =
```
Comment (bastjan) [inline component/main.jsonnet:276]: Key is unused here i would prefer the more generic signature `makeNamespaced(obj)` and further down using an anonymous function.

```jsonnet
std.mapWithKey(
  function(key,obj) makeNamespaced(obj),
  {
```

**Review by bastjan — APPROVED** [review]
LGTM. nice toml renderer 👀

The one nit is just personal preference.

File: component/main.jsonnet (hunk 1 above)
Comment (simu) [reply component/main.jsonnet]: changed as suggested in latest version

File: class/defaults.yml [hunk 2]
```diff
@@ -1,4 +1,57 @@
...
+      general:
+        # the domain which the dns server serves
+        domain: acme-dns.example.com
+        # the hostname of the dns server
+        nsname: acme-dns-ns.example.com
+        nsadmin: support@example.com
+      logconfig:
+        loglevel: info
+        logtype: stdout
+        logformat: text
```
Comment (corvus-ch) [inline class/defaults.yml:42]: Is there a structured log output like json available? If so, please use that one.

File: docs/modules/ROOT/pages/how-tos/setup.adoc [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+
+First, you need to get a local copy of the catalog for the cluster on which you want to deploy acme-dns.
+As mentioned in the prerequisites, acme-dns requires a service of type `LoadBalancer` for the nameserver endpoint.
+
+[source,console]
+----
+$ TMPDIR=$(mktmp -d)
+$ pushd "${TMPDIR}"
+$ export CLUSTER_ID=c-cluster-id <1>
+$ commodore catalog compile "${CLUSTER_ID}"
```
Comment (corvus-ch) [inline docs/modules/ROOT/pages/how-tos/setup.adoc:24]: It proably would be sufficient to assume a reader of this guide knows how to compile a cluster catalog. I suggest to simplify this guide accordingly.

File: docs/modules/ROOT/pages/references/parameters.adoc [hunk 4]
```diff
@@ -2,18 +2,243 @@
...
+
+The acme-dns log configuration.
+
+=== Final configuration
+
+The component merges the contents of `config` into the following default TOML configuration:
+
+[source,toml]
+----
+[general]
```
Comment (corvus-ch) [inline docs/modules/ROOT/pages/references/parameters.adoc:126]: This is an awefull lot to maintain. How do we prevent this from getting out of sync with the coded logic?

**Review by corvus-ch — APPROVED** [review]
I have commented on a few things I find concerning. However, the overal is locking good and I do not have objections to go ahead.

File: docs/modules/ROOT/pages/references/parameters.adoc (hunk 4 above)
Comment (simu) [reply docs/modules/ROOT/pages/references/parameters.adoc:126]: The same way we prevent everything from getting out of sync between the docs and the code, the "have you updated the docs?" checkbox in the PR.

IMHO, this is a tradeoff between good documentation and maintenance overhead, where in this case, I decided that the maintenance overhead is probably acceptable in order to have documentation that clearly describes how acme-dns is configured out of the box.

File: docs/modules/ROOT/pages/how-tos/setup.adoc (hunk 3 above)
Comment (simu) [reply docs/modules/ROOT/pages/how-tos/setup.adoc]: updated in latest version

**Pull request merged**